src/main/java/org/broadinstitute/hellbender/utils/read/BDGAlignmentRecordToGATKReadAdapter.java

```diff
@@ -0,0 +1,40 @@
+package org.broadinstitute.hellbender.utils.read;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import org.bdgenomics.adam.converters.AlignmentRecordConverter;
+import org.bdgenomics.adam.models.SAMFileHeaderWritable;
+import org.bdgenomics.formats.avro.AlignmentRecord;
+
+import java.util.UUID;
+
+/**
+ * Implementation of the {@link GATKRead} interface for the {@link AlignmentRecord} class.
+ *
+ * This adapter wraps an {@link AlignmentRecord} by first converting to a {@link SAMRecord} and then passing through
+ * to {@link SAMRecordToGATKReadAdapter}.
+ */
+public final class BDGAlignmentRecordToGATKReadAdapter extends SAMRecordToGATKReadAdapter {
+    private static final long serialVersionUID = 1L;
+
+    private final AlignmentRecord alignmentRecord;
+
+    public BDGAlignmentRecordToGATKReadAdapter( final AlignmentRecord alignmentRecord) {
+        this(alignmentRecord, null);
+    }
+
+    public BDGAlignmentRecordToGATKReadAdapter( final AlignmentRecord alignmentRecord, final SAMFileHeaderWritable header) {
```

Comment:
It should take the `SAMFileHeaderWritable` as the argument. You'll still call `new AlignmentRecordConverter().convert(alignmentRecord, header)`.

Looking at this again. I think we'll also want a static function that takes an `AlignmentRecord ` and `SAMFileHeaderWritable` and returns a  `SAMRecordToGATKReadAdapter`. In cases where we don't _need_ to keep around the alignmentRecord, it would be best to discard it.